### ISSUE TITLE
Hamburger Menu

### ISSUE THREAD

**Issue by JPRuehmann** [cleidigh/printing-tools-ng#230]
Is there a Way to disable ore move the Entry at the end directly under the Close Entry of the Hamburger Menu?

**Comment by cleidigh** [comment]
@JPRuehmann 
You are correct to ask, this is not the place for the item.
There is a missing tools menu handling in the api. I am waiting for that as trying to hack it will cause problems. I may see about removing temporarily now that I have the button.
@cleidigh

**Comment by Przbl** [comment]
I guess this problem also covers the position of the menu item in the standard menu bar? Right now it is the last item in the File menu and should be moved up if possible (so that Quit is at the bottom).

**Comment by cleidigh** [comment]
@JPRuehmann 
@Przbl 
@DenB10 
@kiki-ja 
@tidebre67 
@RRoenn 
@KC-T
@madvinegar
@Jayonas

Quick poll: 
I am thinking of removing the print ng entry from the Hamburger menu.
-It sits in the wrong place below Quit
- There is not a system print there so why should we be there 
- We can print from many more appropriate places

Votes?
@cleidigh

**Comment by madvinegar** [comment]
I didn't even know there was an NG entry there. I agree to remove it. Wrong place and noone goes through the hamburger menu to print.

**Comment by cleidigh** [comment]
> I didn't even know there was an NG entry there. I agree to remove it. Wrong place and noone goes through the hamburger menu to print.
Akrivos %-}

**Comment by tidebre67** [comment]
@cleidigh 
never used PTNG from the Hamburger Menu. 
So nothing against to remove it there.

**Comment by KC-T** [comment]
Thunderbird Supernova has reduced the two-row horizontal menu to one row leaving not enough space to accommodate so many buttons. The hamburger menu is not really useful for quick access. Putting PTNG at the bottom there is also not useful nor necessary. I only use Ctrl-P. I just removed PTNG from the horizontal menu. Two items for PTNG and its setting still exist at the "File" drop down menu. Fine to change as just mentioned.

Incidentally, when I click the default Print item there, PTNG format does not apply.. When  I click the PTNG item, PTNG format applies. Both have the short cut key Ctrl-P, but the default Print is overridden by PTNG. I did not know this until now. No problem.

**Comment by cleidigh** [comment]
@KC-T
I am fixing the File menu as well.
- moving Print NG and options under Print
- Removing Ctrl-P from TB print
- Shortcut still reassigned to ptng 

@cleidigh

**Comment by Jayonas** [comment]
Although I think that I noticed it in the hamburger menu at one point, I too never actually use that to print.  I always use a toolbar button, either the main toolbar or whatever you call the smaller toolbar on the right side of the header area in the email reading view.  I agree that the item doesn't really feel like it belongs in the hamburger menu, especially not way at the bottom, and I also agree that it generally makes the most sense for NG to appear in the same places that the built-in Print appears.

**Comment by kiki-ja** [comment]
@cleidigh

I agree to the removal with the condition that the PTNG toolbar icons and PTNG context menu remain as they are.
The reasons are as follows.

1. Thunderbird the Hamburger menu has Print by default.
2. Users who use it can also choose to print by PTNG. 
3. On the other hand, the design of the Hamburger menu is complicated and it is difficult to add or change the position of the PTNG print menu.
Note: The design specifications do not taken into account for customization by add-ons.
4. Even if removed, access to PTNG printing is still provided via toolbar icons and context menus.

Other:
Please do not remove the Ctrl+Shift+P (PTNG Option) and Ctrl+P (PTNG Print) shortcut key.
However, Ctrl+P (PTNG Print) must be assigned to a new shortcut key.

Reason:
Because it is convenient and I myself use it often.
Ctrl+P (Thunderbird default) and Ctrl+P (PTNG Print) are convenient because it is possible to use them in different ways.

@kiki-ja

**Comment by cleidigh** [comment]
@kiki-ja 
Ok I had not thought about creating a substitute shortcut for the native print.
Perhaps Control-Alt-P ?

I will do this for b2, but we can always change.
@cleidigh

**Comment by madvinegar** [comment]
If we could remove the oem "print" option from the file menu, it would be perfect. :)

Some times I get confused and I press it instead of Print_NG.

![1](https://github.com/cleidigh/printing-tools-ng/assets/11718763/9c48463e-9254-41e9-8028-cc9c4d8018d2)

**Comment by kiki-ja** [comment]
I disagree removing of Print… (Ctrl+P) from Thunderbird's default menu.
The reasons are clear.

1. Add-ons are functions to extend, not to remove the main functions.
2. If you want to customize (remove or hide) Thunderbird's default menu, you should use userChrome.css or other add-ons.
3. Thus, it is not the role of PrintingTools NG.

@kiki-ja

**Comment by madvinegar** [comment]
I thought it was like that in previous versions. (i.e. in Print_NG v2.xx and in thunderbird v.102.x.x.

In these versions you only had the print_Ng option in the file menu.

Am I wrong...?

**Comment by KC-T** [comment]
I have been using Ctrl-P for Printing Tools NG and prefer retaining it to override the default Print, which can also be retained in the File menu in case people would like to use the default, which should not be very frequent to deserve two different short cut keys.

**Comment by cleidigh** [comment]
@madvinegar 
You are correct in pre v115 versions the File entry for the native print was removed.
Because of the new menu api I have not done any manipulation other than a couple of specific attributes. Besides that I also think per @kiki-ja comment that maintaining the native print is more appropriate so users can use it if desired.

It is not perfect, but we have to deal with compromises.
What I really hate is the fact that extensions are forced to the bottom of the menus.
@cleidigh

**Comment by DecoDogs** [comment]
TB 115.18.0
PTNG 3.1.0
Linux Mint 21.3

I realize this is an older issue, but it is the only one I found addressing the Control+P topic, so please forgive me if this is an inappropriate place to comment.

I've just discovered PrintingTools NG and think it will be quite useful.  But, I want to use TB's native printing facility much of the time, and, while I can get to that via File -> Print, it is just a bit cumbersome.

I found the entries in the code where the remapping is being done, but have no idea the degree of difficulty in implementing what I've mentioned.

So, is there any thought to allowing the user to select how Control+P works?

Thank you

**Comment by KC-T** [comment]
I have been happily using PTNG Ctrl-P, same as what I commented on Oct 30, 2023.

Not sure how you find PTNG useful but still want to use the default File --> Print for Thunderbird default printing. Perhaps, you can still use PTNG Ctrl-P but config PTNG to change the Thunderbird default layout and format only to the extent you like.

**Comment by kiki-ja** [comment]
@DecoDogs

>I've just discovered PrintingTools NG and think it will be quite useful. 

I am glad to see that more and more users are using PrintingTools NG.

>But, I want to use TB's native printing facility much of the time, and, while I can get to that via File -> Print, it is just a bit cumbersome.

There are two other methods besides Print in the File menu. You know them, don't you?
　・Right click on selected message -> Print...
　・One-click on the Print Icon button (Right click on the toolbar -> Customize... -> Place Print Icon where you want it).

>So, is there any thought to allowing the user to select how Control+P works?

If you find the above access method cumbersome, I recommend that you submit an enhancement to "Bugzilla" since the native print menu is a specification of Thunderbird itself.

Bugzilla Main Page
https://bugzilla.mozilla.org/home

@kiki-ja

**Comment by DecoDogs** [comment]
@KC-T 
For some things, I like the much cleaner page layout capable with PTNG (I've hidden all the standard headers, and the title [&T]).  But other times, I want to see the 'usual'' output with headers - typical email contents.  Does PTNG have the concept of 'profiles' and I've just missed it?

@kiki-ja 
I had completely forgotten about customizing the toolbar!  I've done that in both TB and FF, but it's been awhile and it just didn't occur to me.  So thanks!

The right-click -> print I had found.  I'll just try both ways to see which works best for me.

That's a good thought regarding filing and enhancement, though I have a feeling it would just be a waste of time!

Thanks to you both!

**Comment by kiki-ja** [comment]
@DecoDogs

>For some things, I like the much cleaner page layout capable with PTNG (I've hidden all the standard headers, and the title [&T]). But other times, I want to see the 'usual'' output with headers - typical email contents.

You wish to use multiple print patterns (page layout), i.e., use them together.
Unfortunately, currently you have to choose between “native” print or “PrintingTools NG” print.
It might be useful if these could be integrated and multiple print patterns could be preset in PrintingTools NG.


>Does PTNG have the concept of 'profiles' and I've just missed it?

PrintingTools NG does not have the concept of a dedicated profile.
The configuration values for PrintingTools NG are written in Thunderbird's prefs.js file, which stores the overall configuration values.

prefs.js can be found by opening Setting -> General -> Config Editor...(about:config).
The configuration values for PrintingTools NG are described by parameters starting with “extensions.printingtoolsng".

**Thereafter, please post a new issue if you have any questions or requests (or wishes).**

@kiki-ja

**Comment by cleidigh** [comment]
@DecoDogs 
One way to get a separate hotkey for the standard print might be to us the tbkeys addon. Check with them to see if they can do it.
https://addons.thunderbird.net/en-US/thunderbird/addon/tbkeys-lite/?src=ss

@cleidigh

**Comment by DecoDogs** [comment]
@cleidigh

Thanks.
I had noticed that addon, but just haven't had time to work with it yet.
